client/rpc-servers/src/lib.rs

```diff
@@ -68,6 +69,40 @@ pub fn rpc_handler<M: PubSubMetadata>(
 	io
 }
 
+/// RPC server-specific prometheus metrics.
+#[derive(Debug, Clone)]
+pub struct ServerMetrics {
+	/// Number of sessions opened.
+	session_opened: Option<Gauge<U64>>,
+	/// Number of sessions closed.
+	session_closed: Option<Gauge<U64>>,
+}
+
+impl ServerMetrics {
+	/// Create new WebSocket RPC server metrics.
+	pub fn new(registry: Option<&Registry>) -> Result<Self, PrometheusError> {
+		registry.map(|r| Ok(Self {
+			session_opened: register(
+				Gauge::new(
+					"rpc_sessions_opened",
+					"Number of persistent RPC sessions opened",
```

Comment:
in general it is more suitable to use a `counter` instead of a `gauge` in case the value only ever is supposed to increase. this will make prometheus interpret resets in the "right" way.

Reply:
Good point, fixed, thanks!